daemon/containerd/image.go

```diff
@@ -158,15 +163,17 @@ func (i *ImageService) GetImageManifest(ctx context.Context, refOrID string, opt
 	}
 
 	if containerdimages.IsManifestType(desc.MediaType) {
+		if options.Platform != nil && desc.Platform == nil {
```

Comment:
ISTR historically, "no platform" meant "linux/amd64". Reason for that was that platform didn't exist yet, and before that "linux/amd64" was the only option. Because of that, a missing platform had to be considered `linux/amd64`


Not sure if that's something we should still take into account though. @tianon @cpuguy83 any thoughts?

Reply:
Hm good to know thanks, I'll dig a little bit more

Reply:
I don't think it's a blocker, but yes, we should dig up some of those decisions and find a place where to document them. They're the "odd quirks" kinda things that usually weren't documented (beyond possibly a comment on a PR)

Reply:
I don't think we should treat `desc.Platform == nil` as a platform mismatch. The `platform` property is optional and `SHOULD be present if its target is platform-specific.`. So it's perfectly fine for it to be nil and that should be interpreted as "any platforms".